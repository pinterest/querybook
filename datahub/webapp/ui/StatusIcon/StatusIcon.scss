.StatusIcon {
    color: var(--color-null);
    margin: 1.5px 5px 1.5px 0px;
    i {
        vertical-align: middle;
    }

    &.success {
        color: var(--color-true);
    }
    &.warning {
        color: var(--color-yellow);
    }
    &.error {
        color: var(--color-false);
    }
    &.running {
        color: var(--color-blue);
    }
}
