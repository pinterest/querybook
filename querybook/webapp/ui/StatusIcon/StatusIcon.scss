.StatusIcon {
    display: inline-flex;
    color: var(--color-null);
    margin-right: 6px;
    i {
        vertical-align: middle;
    }

    &.success {
        color: var(--color-true);
    }
    &.warning {
        color: var(--color-yellow);
    }
    &.error {
        color: var(--color-false);
    }
    &.running {
        color: var(--color-blue);
    }
}
