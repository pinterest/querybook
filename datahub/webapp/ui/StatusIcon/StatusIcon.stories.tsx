import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { StatusIcon, IStatusIconProps } from './StatusIcon';
import { Status } from 'const/queryStatus';

export const _StatusIcon = (args: IStatusIconProps) => (
    <div className="flex-column">
        <div>
            <StatusIcon {...args} status={Status.success} /> Success
        </div>
        <div>
            <StatusIcon {...args} status={Status.warning} /> Warning
        </div>
        <div>
            <StatusIcon {...args} status={Status.error} /> Error
        </div>
        <div>
            <StatusIcon {...args} status={Status.running} /> Running
        </div>
        <div>
            <StatusIcon {...args} status={Status.none} /> None
        </div>
    </div>
);

_StatusIcon.args = {
    tooltip: 'Some tooltip',
    hollow: false,
};

export default {
    title: 'Stateless/StatusIcon',
    decorators: [centered],
};
