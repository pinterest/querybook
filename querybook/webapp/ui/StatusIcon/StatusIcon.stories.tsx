import centered from '@storybook/addon-centered/react';
import React from 'react';

import { Status } from 'const/queryStatus';

import { IStatusIconProps, StatusIcon } from './StatusIcon';

export const _StatusIcon = (args: IStatusIconProps) => (
    <div className="flex-column">
        <div>
            <StatusIcon {...args} status={Status.success} /> Success
        </div>
        <div>
            <StatusIcon {...args} status={Status.warning} /> Warning
        </div>
        <div>
            <StatusIcon {...args} status={Status.error} /> Error
        </div>
        <div>
            <StatusIcon {...args} status={Status.running} /> Running
        </div>
        <div>
            <StatusIcon {...args} status={Status.none} /> None
        </div>
    </div>
);

_StatusIcon.args = {
    tooltip: 'Some tooltip',
    hollow: false,
};

export default {
    title: 'Stateless/StatusIcon',
    decorators: [centered],
};
