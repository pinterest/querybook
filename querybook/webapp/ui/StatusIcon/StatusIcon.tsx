import React from 'react';
import clsx from 'clsx';

import { Status } from 'const/queryStatus';

import './StatusIcon.scss';

export interface IStatusIconProps {
    status: Status;
    tooltip?: string;
    hollow?: boolean;
}

export const StatusIcon: React.FunctionComponent<IStatusIconProps> = ({
    status,
    tooltip,
    hollow,
}) => {
    const iconProps = {
        status,
        className: clsx({
            fas: !hollow,
            far: hollow,
            'fa-circle': true,
        }),
    };

    const spanProps = {
        className: clsx({
            StatusIcon: true,
            [status]: true,
        }),
    };
    if (tooltip) {
        spanProps['aria-label'] = tooltip;
        spanProps['data-balloon-pos'] = 'up';
    }

    return (
        <span {...spanProps}>
            <i {...iconProps} />
        </span>
    );
};

StatusIcon.defaultProps = {
    status: Status.none,
};
