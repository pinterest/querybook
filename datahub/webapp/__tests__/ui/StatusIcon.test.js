import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { StatusIcon } from '../../ui/StatusIcon/StatusIcon';

it('renders without crashing', () => {
    shallow(<StatusIcon status="success" />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        let wrapper = shallow(<StatusIcon status="success" />);
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot', () => {
        let wrapper = shallow(<StatusIcon status="warning" />);
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<StatusIcon status="success" />);
        expect(output).toMatchSnapshot();
    });
});
