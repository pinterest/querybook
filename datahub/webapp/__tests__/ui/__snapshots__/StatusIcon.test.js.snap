// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<span
  className="StatusIcon success"
>
  <i
    className="fas fa-circle"
    status="success"
  />
</span>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<span
  className="StatusIcon success"
>
  <i
    className="fas fa-circle"
    status="success"
  />
</span>
`;
